src/grocker/utils.py

```diff
@@ -44,23 +45,19 @@ def unit_list(l):
     return digest.hexdigest()
 
 
-def default_image_name(config, release):
-    req = pkg_resources.Requirement.parse(release)
-    if not str(req.specifier).startswith('=='):
-        raise RuntimeError("Only fixed version can use default image name.")
-
+def default_image_name(config, req):
     docker_image_prefix = config['docker_image_prefix']
     if config['image_base_name']:
         img_name = config['image_base_name']
     elif req.extras:
         img_name = "{project}-{extra_requirements}".format(
-            project=req.project_name,
+            project=packaging.utils.canonicalize_name(req.project_name),
```

Comment:
`packaging.utils.canonicalize_name` is already call in `GrockerRequirement` (same comment on line 58)